src/snarky-typescript/src/index.ts

```diff
@@ -0,0 +1,517 @@
+interface ToFieldElements<Field, Type> {
+  toFieldElements(this: Type): Field[];
+}
+
+interface OfFieldElements<Field, Type> {
+  ofFieldElements(elts: Field[]): Type;
+}
+
+interface AsFieldElements<Field, Type>
```

Comment:
I'm not sure `AsFieldElements` is a very descriptive name for this (when I read this, it sounds like a synonym of `toFieldElements`, not an interface which can bidirectionally convert to and from field elements). Maybe `FieldElementsSerializable` or something even more abstract such as `SnarkyValue` might be more descriptive. If you rename the sum type below, you could even call this `FieldConvertible`.

Reply:
I think Rust's `To` and `From` are simpler than OCaml's `To` and `Of` as well.